src/main/java/org/broadinstitute/consent/http/service/dao/DraftServiceDAO.java

```diff
@@ -0,0 +1,156 @@
+import org.broadinstitute.consent.http.models.DraftInterface;
+import org.broadinstitute.consent.http.models.DraftSummary;
+import org.broadinstitute.consent.http.models.FileStorageObject;
+import org.broadinstitute.consent.http.models.User;
+import org.glassfish.jersey.media.multipart.FormDataBodyPart;
+import org.jdbi.v3.core.Jdbi;
+
+public class DraftServiceDAO {
+
+  private final Jdbi jdbi;
+  private final DraftDAO draftDAO;
+  private final DraftFileStorageServiceDAO draftFileStorageServiceDAO;
+
+  @Inject
+  public DraftServiceDAO(Jdbi jdbi, DraftDAO draftDAO,
+      DraftFileStorageServiceDAO draftFileStorageServiceDAO) {
+    this.jdbi = jdbi;
+    this.draftDAO = draftDAO;
+    this.draftFileStorageServiceDAO = draftFileStorageServiceDAO;
+  }
+
+  public void insertDraft(DraftInterface draft)
+      throws SQLException, BadRequestException {
+    jdbi.useHandle(handle -> {
+      handle.getConnection().setAutoCommit(false);
+      try {
+        draftDAO.insert(draft.getName(), draft.getCreateDate().toInstant(),
+            draft.getCreateUser().getUserId(), draft.getJson(), draft.getUUID(),
+            draft.getType().getValue());
+      } catch (Exception e) {
+        handle.rollback();
+        throw new BadRequestException(
+            "Error submitting draft.  Drafts require valid json to be submitted.");
+      }
+      handle.commit();
+    });
+  }
+
+  public DraftInterface updateDraft(DraftInterface draft, User user) throws SQLException {
+    draft.setUpdateUser(user);
+    draft.setUpdateDate(new Date());
+    jdbi.useHandle(handle -> {
+      handle.getConnection().setAutoCommit(false);
+      try {
+        draftDAO.updateDraftByDraftUUID(draft.getName(),
+            draft.getUpdateDate().toInstant(), draft.getUpdateUser().getUserId(), draft.getJson(),
+            draft.getUUID(), draft.getType().getValue());
+      } catch (Exception e) {
+        handle.rollback();
+      }
+      handle.commit();
+    });
+    return getAuthorizedDraft(draft.getUUID(), user);
+  }
+
+  public DraftInterface getAuthorizedDraft(UUID draftUUID, User user) {
+    DraftInterface draft;
+    try {
+      draft = findDraftByDraftUUID(draftUUID);
+    } catch (SQLException e) {
+      throw new NotFoundException(
+          String.format("Draft with UUID %s not found.", draftUUID.toString()));
+    }
+    if (Objects.isNull(draft)) {
+      throw new NotFoundException(
+          String.format("Draft with UUID %s not found.", draftUUID.toString()));
+    }
+    if (!user.getUserId().equals(draft.getCreateUser().getUserId()) && !user.hasUserRole(
+        UserRoles.ADMIN)) {
+      throw new NotAuthorizedException("User not authorized to modify resource.");
+    }
+    return draft;
+  }
+
+  public void deleteDraftsByUser(User user) {
+    Set<DraftInterface> userDrafts = findDraftsForUser(user);
+    for (DraftInterface draft : userDrafts) {
+      deleteDraft(draft, user);
+    }
```

Comment:
In general I would avoid using `Set` unless the code requires it, in case the collection type changes in the future. You could use `Collection<DraftInterface` here or use `forEach()` to write it functionally
```suggestion
    findDraftsForUser(user).forEach(draft -> deleteDraft(draft, user));
```

Another way to solve this is to use `var`.

Reply:
Moved to Collection to be consistent with other changes.